bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/handler/BroadlinkRemoteHandler.java

```diff
@@ -0,0 +1,496 @@
+                    new StringType("Error found during when entering IR learning mode"));
+            logger.warn("Exception while sending command", e);
+        }
+
+        return null;
+    }
+
+    protected void sendCode(byte[] code) {
+        logger.debug("Sending code: {}", Utils.toHexString(code));
+        sendCommand(COMMAND_BYTE_SEND_CODE, code, "send remote code");
+    }
+
+    protected ByteArrayOutputStream buildCommandMessage(byte commandByte, byte[] codeBytes) throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        byte[] preamble = new byte[4];
+        preamble[0] = commandByte;
+        outputStream.write(preamble);
+        if (codeBytes.length > 0) {
+            outputStream.write(codeBytes);
+        }
+
+        return outputStream;
+    }
+
+    protected byte[] extractResponsePayload(byte[] responseBytes) throws IOException {
+        byte decryptedResponse[] = decodeDevicePacket(responseBytes);
+        // Interesting stuff begins at the fourth byte
+        return Utils.slice(decryptedResponse, 4, decryptedResponse.length);
+    }
+
+    private void handleIRCommand(String irCommand, boolean replacement) {
+        try {
+            String message = "";
+            if (replacement) {
+                message = "Modifying ";
+            } else {
+                message = "Adding ";
+            }
+
+            BroadlinkMappingService mappingService = this.mappingService;
+            if (mappingService == null) {
+                logger.warn("Mapping service is null, this should not happen");
+                updateState(BroadlinkBindingConstants.LEARNING_CONTROL_CHANNEL, new StringType("NULL"));
+                return;
+            }
+
+            updateState(BroadlinkBindingConstants.LEARNING_CONTROL_CHANNEL,
+                    new StringType(message + irCommand + "..."));
+
+            byte[] response = sendCommand(COMMAND_BYTE_CHECK_LEARNT_DATA, "send learnt code check command");
+
+            if (response == null) {
+                logger.warn("Got nothing back while getting learnt code");
+                updateState(BroadlinkBindingConstants.LEARNING_CONTROL_CHANNEL, new StringType("NULL"));
+            } else {
+                String hexString = Utils.toHexString(extractResponsePayload(response));
+                String cmdLabel = null;
+                if (replacement) {
+                    cmdLabel = mappingService.replaceCode(irCommand, hexString, CodeType.IR);
+                    message = "modified";
+                } else {
+                    cmdLabel = mappingService.storeCode(irCommand, hexString, CodeType.IR);
+                    message = "saved";
+                }
+                if (cmdLabel != null) {
+                    logger.info("BEGIN LAST LEARNT CODE");
+                    logger.info("{}", hexString);
+                    logger.info("END LAST LEARNT CODE ({} characters)", hexString.length());
```

Comment:
In the past people had to copy & paste these codes into a mapping file. However, this is no longer required any more and it can be done through the UI instead.